app/notifications/rest.py

```diff
@@ -48,16 +53,30 @@ def get_notification_by_id(notification_id):
         )
         notification.to = recipient
         notification.normalised_to = recipient
-    return (
-        jsonify(
-            data={
-                "notification": notification_with_personalisation_schema.dump(
-                    notification
+
+    if not hasattr(notification, "body") or not notification.body:
+        template_dict = template_model_to_dict(notification.template)
+        template = get_template_instance(
+            template_dict, notification.personalisation or {}
+        )
+        notification.body = template.content_with_placeholders_filled_in
+        if hasattr(template, "subject"):
+            try:
+                notification.__dict__["subject"] = template.subject
```

Comment:
Is there a reason we have to reach into the internal object attributes to set this vs. just `notification.subject` - which I know doesn't actually exist on a `Notification` object.  This makes me wonder if the work being done in this `if` block should actually be operating against the `schema` down below, which _does_ include these attributes (pulled from the `TemplateHistorySchema` - unless that got changed with this shift to this new schema?